js/script.js

```diff
@@ -562,77 +488,94 @@ async function reset() {
  * Click handler for the connect/disconnect button.
  */
 async function clickConnect() {
-    if (espTool.connected()) {
-        // oops we're supposed to be disconnected already
-        await disconnect();
-    }
+    await disconnect();
 
-    // we have to connect and sync before we can do anything useful
-    await connect();
-    if (!await espTool.sync()) { throw "Synchronization Failure" }
-    // connection is good at this point
+    butConnect.textContent = "Connecting...";
+    butConnect.disabled = true
+  
+    const esploaderMod = await esptoolPackage;
```

Comment:
It came from https://github.com/NabuCasa/esp-web-flasher/blob/main/js/script.js#L183, but I don't think it's necessary.

Reply:
I should be specific, it looks like it's working at first, but the connection function fails.

Reply:
The reason for this is that for `esp-web-flasher` we dynamically decide the import inside the `index.html` to point at dev during development, while pointing at NPM on GitHub pages.

[It uses a dynamic import (`import()`).](https://github.com/NabuCasa/esp-web-flasher/blob/main/index.html#L38) That returns a promise that we await in `script.js`. 

Because you import `script.js` with `type=module`, you can either do the dynamic import in your `script.js` or host the file in the same directory and do `import * as esploader from "./esploader.js";`